builtin/game/misc.lua

```diff
@@ -87,12 +87,14 @@ end
 local player_list = {}
 
 core.register_on_joinplayer(function(player)
+	if minetest.is_singleplayer() then return end
 	local player_name = player:get_player_name()
 	player_list[player_name] = player
```

Comment:
Actually, are you sure the player_list isn't needed? `get_connected_players` might be used by mods and likely needs the user to be in that list, even in singleplayer.